aiida/calculations/transfer.py

```diff
@@ -0,0 +1,250 @@
+    new folder in the remote machine (RemoteData) and put all the files there and will either:
+
+        (1) leave them there (``retrieve_files = False``) or ...
+        (2) create a local FolderData and copy there all the files (``retrieve_files = True``)
+
+    The `instructions` dict must also contain at least one list with specifications of which files
+    to copy and from where. All these lists take tuples of 3 that have the following format:
+
+    .. code-block:: python
+
+        [ ..., ( source_node_key, path_to_file_in_source, path_to_file_in_target), ... ]
+
+    where the ``source_node_key`` has to be the respective one used when providing the node in the
+    ``source_nodes`` input nodes dictionary.
+
+
+    The two main lists to include are ``local_files`` (for files to be taken from FolderData nodes)
+    and ``remote_files`` (for files to be taken from RemoteData nodes). Alternatively, files inside
+    of RemoteData nodes can instead be put in the ``symlink_files`` list: the only difference is that
+    files from the first list will be fully copied in the target RemoteData folder, whereas for the
+    files in second list only a symlink to the original file will be created there. This will only
+    affect the content of the final RemoteData target folder, but in both cases the full file will
+    be copied back in the local target FolderData (if ``retrieve_files = True``).
+    """
+
+    @classmethod
+    def define(cls, spec):
+        super().define(spec)
+
+        spec.input(
+            'instructions',
+            valid_type=orm.Dict,
+            help='A dictionary containing the `retrieve_files` flag and at least one of the file lists:'
+            '`local_files`, `remote_files` and/or `symlink_files`.',
+            validator=validate_instructions,
+        )
+        spec.input_namespace(
+            'source_nodes',
+            valid_type=(orm.FolderData, orm.RemoteData),
+            dynamic=True,
+            help='All the nodes that contain files referenced in the instructions.',
+        )
+
+        # The transfer just needs a computer, the code are resources are set here
+        spec.inputs['code'].required = False
+        spec.inputs['metadata']['computer'].required = True
+        spec.inputs['metadata']['options']['resources'].default = {
+            'num_machines': 1,
+            'num_mpiprocs_per_machine': 1,
+        }
+
+        spec.inputs.validator = validate_transfer_inputs
+
+    def prepare_for_submission(self, folder):
+        source_nodes = self.inputs.source_nodes
+        instructions = self.inputs.instructions.get_dict()
+
+        local_files = instructions.get('local_files', [])
+        remote_files = instructions.get('remote_files', [])
+        symlink_files = instructions.get('symlink_files', [])
+        retrieve_files = instructions.get('retrieve_files')
+
+        calc_info = CalcInfo()
+        calc_info.skip_submit = True
+        calc_info.codes_info = []
```

Comment:
does this need to be set? Or can we omit this line?

Reply:
You mean `calc_info.codes_info = []`? I get an error when I remove it (it is a bit long log so I'm not sure it makes sense to copy here; I can't really identify the specific location of the error from the log).

Reply:
Same here, instead of starting to refactor the `presubmit` let's leave this for now